### PR TITLE
WIP: Add travis and a first test

### PR DESCRIPTION
**Pull request by pcsforeducation**

### CODE REVIEW

**Comment by aweeks** [comment]
+1

File: agent_test.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	"testing"
+)
+
+func testServerAndClient(code int, body string) (*httptest.Server, *IronicAPIClient) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(code)
+		fmt.Fprintln(w, body)
+	}))
+	client := &IronicAPIClient{
+		server.URL + "/", "agent_ipmi", &http.Client{},
```
Comment (russellhaering) [inline agent_test.go]: I _think_ these values should be on separate lines, but I'll defer to `gofmt`

**Comment by russellhaering** [comment]
Other than formatting, this looks good.

**Comment by pcsforeducation** [comment]
gofmt didn't fix complain about them being on the same line, but it didn't complain if they were on separate either, so I put them on separate lines.

**Pull request closed without merging**